test_federated_learning.ipynb

```diff
@@ -0,0 +1,255 @@
+{
```

Comment:
I am used to having a rationale/TLDR on top of notebooks so if you ever go back to it you still know what the purpose of it was and what it does. Would be nice to add that here?

Reply:
Added it